### PR TITLE
WIP: Refactoring preview builds GitHub action

### PR DESCRIPTION
**Pull request by lefherz**
Thanks @arnested for this suggestion!

reference: https://github.community/t5/GitHub-Actions/Publishing-download-links-in-the-check-details-instead-of/m-p/37273/highlight/false#M2880

### CODE REVIEW

**Comment by lefherz** [comment]
@Simon-Laux refactored it a bit, as you complained about code prettiness ;) do we actually have coding guidelines somewhere? e.g. I didn't respect the 80 chars max screen width I know from python.

**Comment by Jikstra** [comment]
> @Simon-Laux refactored it a bit, as you complained about code prettiness ;) do we actually have coding guidelines somewhere? e.g. I didn't respect the 80 chars max screen width I know from python.

For js we use the standard guidelines. You can start the linter by `npx standard`

File: .github/workflows/build-preview.yml [hunk 1]
```diff
@@ -61,32 +58,39 @@ jobs:
...
         curl -f -s -X POST --header "authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" --url "$GITHUB_API_URL" --header "content-type: application/json" --data "$STATUS_DATA"
-        export STATUS_DATA="{\"state\": \"success\", \"description\": \"Download it here:\", \"context\": \"Windows Preview Build\", \"target_url\": \"${EXE_URL}\"}"
+        # Post Windows download link to check details
+        export STATUS_DATA="{\"state\": \"success\", \
+                             \"description\": \"Download it here:\", \
+                             \"context\": \"Windows Preview Build\", \
+                             \"target_url\": \"${EXE_URL}\"}"
         curl -f -s -X POST --header "authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" --url "$GITHUB_API_URL" --header "content-type: application/json" --data "$STATUS_DATA"
+        if md5sum -c pkgsum --status; then 
+            export STATUS_DATA="{\"state\": \"failed\", \  # this value is probably not valid
```
Comment (Simon-Laux) [inline .github/workflows/build-preview.yml]: `# this value is probably not valid`
resolve this comment before merge?

Comment (lefherz) [reply .github/workflows/build-preview.yml]: I don't have time before next week, but yes, I will resolve this.

**Comment by lefherz** [comment]
Isn't the point that the package-lock.json is changed through npm install? and that it's concerning if it doesn't change?

In that case we could just save the first sum after the `change build name and Product Name to development id` step instead of beforehand. But maybe I got it totally wrong, I have no experience with npm at all, and pretty much guessed what kind of action you need^^

**Comment by Simon-Laux** [comment]
> Isn't the point that the package-lock.json is changed through npm install? and that it's concerning if it doesn't change?

its actually the other way around - it must not change.

**Comment by Simon-Laux** [comment]
@missytake whats the progress of this pr? Is it done and we just need to resolve the conflicts?

**Comment by missytake** [comment]
quite far iirc. I can look into it over the weekend

**Comment by lefherz** [comment]
Hm, for some reasons the package-lock file I build locally and the package-lock file the check-lock action builds differ, even though we use the same node version o.0 any explanations?

```
v10.17.0
Checksum before npm i:
85466864afe574c48a03c5986020eab6 package-lock.json
Checksum after npm i:
be99f5b99d46749eba83e079109dea30  package-lock.json
```

**Pull request closed without merging**